apps/docs-website/blog/2025-01-5-inversion-of-control.mdx

```diff
@@ -0,0 +1,162 @@
+            message: "...",
+        },
+        {
+            selector: "ForInStatement",
+            message: "...",
+        },
+    ],
+    allows: [
+        "LabeledStatement",
+        "ForInStatement",
+    ],
+},
+```
+
+In the `adjuncts` array, the user can add new syntax features to disallow on-top of the default ones, while in the `allows` array they can disable some of the default ones.
+
+As you can see, the offered API looked pretty complex, arbitrary, tangled and unintuitive. Both `adjuncts` and `allows` suggest at "adding" something, so it could be confusing to understand "what did what" for newcomers of the library. In short: it was not user-friendly at all.
+
+### The solution
+
+Inversion of control to the rescue.
+
+Instead of wrapping the `no-restricted-syntax` option into the Sheriff options, we let the users define and configure the `no-restricted-syntax` rule directly in their own config and offer them **composables** around it. This way the user is back in the driver seat and has full control again.
+
+Sheriff now expose a variable called `baseNoRestrictedSyntaxRules` that contains the contents of the Sheriff-configured `no-restricted-syntax` rule. So adding new restricted features on top of the default ones is as simple as:
+
+```ts
+no-restricted-syntax: [
+    2,
+    ...baseNoRestrictedSyntaxRules,
+    // your custom rules here...
+],
+```
+
+While disabling some of the default ones is a little more involved.
+
+Every entry prints the index of itself at the end of the message property, so users should use the index to identify which entry they want to remove.
+
+In the docs we suggest using the native Javascript [`toSpliced()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced) method to remove an entry from the `baseNoRestrictedSyntaxRules` array:
+
+```ts
+no-restricted-syntax: [
+    2,
+    ...baseNoRestrictedSyntaxRules.toSpliced(2, 1)
+    // your custom rules here...
+],
+```
+
+[🠪 Learn more about this API in the docs](https://www.eslint-config-sheriff.dev/docs/configuration#customizing-the-no-restricted-syntax).
+
+### Going forward
+
+Recently an ESlint contributor released a very interesting alternative to the `no-restricted-syntax` rule called [eslint-no-restricted](https://github.com/bradzacher/eslint-no-restricted).
+
+Sheriff will most likely adopt this library in the future and abandon the current "homegrown" solution, as it seems to provide a superior DX.
+
+## Typescript-eslint `project` API
+
+Another case where Sheriff was swallowing up complexity at the cost of giving up a level of control to the user was the `parserOptions.project` API.
+
+### The problem
+
+Sheriff was passing the `tsconfig.json` to the ESLint config in a hard-coded way:
+
+```ts
+project: './tsconfig.json'
+```
+
+Naturally users with multiple `tsconfig.json` on the same level were feeling limited, because it wasn't possible to specify different paths for the `tsconfig.json`.
```

Comment:
```suggestion
Naturally, users with multiple `tsconfig.json`s felt limited because it wasn't possible to specify different paths for `tsconfig.json`s.
```